src/Nethermind/Nethermind.Baseline/BaselineTree.cs

```diff
@@ -0,0 +1,253 @@
+using System;
+using System.Runtime.CompilerServices;
+using Nethermind.Core.Crypto;
+using Nethermind.Serialization.Rlp;
+using Nethermind.Trie;
+
+[assembly: InternalsVisibleTo("Nethermind.Baseline.Test")]
+
+namespace Nethermind.Baseline
+{
+    public abstract partial class BaselineTree
+    {
+        private const int LeafRow = 32;
+        private const int LeafLevel = 0;
+        public const int TreeHeight = 32;
+        private const ulong FirstLeafIndexAsNodeIndex = MaxNodes / 2;
+        private const ulong MaxNodes = (1ul << (TreeHeight + 1)) - 1ul;
+        private const ulong MaxNodeIndex = MaxNodes - 1;
+
+        private readonly IKeyValueStore _keyValueStore;
+        private readonly byte[] _dbPrefix;
```

Comment:
I see dbPrefix is used to differentiate in IKeyValueStore. I would prefer that injected IKeyValueStore already handles this in some way (so we inject different one to each tree), similar to ColumnDb for RocksDb. This is kind of abstraction leakage.

Reply:
no, I think this would give  responsibility to IKeyValueStore that it should not have and would break the many small interfces part of SOLID

we could create a prefixable store which would be a decorator for the IKeyValueStore or something but I think it is unnecessary